### DISCUSSION TITLE
How are treatments positioned on the graph?

### DISCUSSION THREAD

**Discussion by JLChamberlain** [NightscoutFoundation/xDrip#2507] [General]
In NS the treatment is always placed on the current BG reading on the graph when the treatment is added. In xDrip the placement of the treatment on the graph seems all over the place. How is the placement decided and how can I just get it to be placed on the current BG reading when adding the treatment (like how it's displayed on NS)?

**Comment by Navid200** [comment]
I could be wrong.  But, we have an open issue that seems to me is caused because in one instance a note is placed on the curve.  But, when the unit is changed, the note goes off screen resulting in the vertical axis to change:  https://github.com/NightscoutFoundation/xDrip/issues/1299

Would you be able to provide a screenshot demonstrating what the problem is?

**Reply by JLChamberlain** [reply]
It's nothing like that. If say my BG is 5.3 and then I add a treatment on NS the treatment will appear on 5.3 at the time of the note whereas in xDrip sometimes the treatment is positioned on the 4.0 line, 10 lines or sometimes on the current BG reading.

**Reply by JLChamberlain** [reply]
<img width="540" alt="Screenshot_20221127-170144" src="https://user-images.githubusercontent.com/22307144/204285738-86dc5c4a-cddd-47d0-8c97-bafc6b17b2bc.png">

If you look at the screenshot I've attached the placement of the treatments makes no sense. Where as on NS the treatment is always placed on the line of the current reading.

**Reply by tolot27** [reply]
I suspect it is simply shifted for readability.

**Reply by steve8x8** [reply]
Hm. I'm using one of the latest nightlies, and have the insulin markers sitting on the 70mg/dL (low margin) line while the meals are marked somewhat higher (around 115 mg/dL, but I vaguely remember that the actual number looked complicated), at fixed heights.
Depending on where your curve dwells it may provide better readability indeed - I always disliked the big blobs NS would show.
BTW, adding notes to individual points will place them nicely above the curve here (by about 30 mg/dL, or 1.7 mmol/L - which is what seems to happen with your I and C markers for a reason unclear to me).
